iphone/cli/commands/_build.js

```diff
@@ -6842,6 +6984,14 @@ iOSBuilder.prototype.invokeXcodeBuild = function invokeXcodeBuild(next) {
 		}
 	}
 
+	if (this.target === 'macos' || this.target === 'dist-macappstore') {
+		args.push('SUPPORTS_MACCATALYST=YES');
+	}
+
+	if (this.target === 'mac') {
+		args.push('SUPPORTS_MACCATALYST=YES');
+	}
+
 	// Exclude arm64 architecture from simulator build in XCode 12+ - TIMOB-28042
 	if (this.target === 'simulator' && parseFloat(this.xcodeEnv.version) >= 12.0) {
 		args.push('EXCLUDED_ARCHS=arm64');
```

Comment:
I had to modify this exclusion. It does need to stick around if we don't want to force all native modules to be rebuilt.

Basically I modified it to record any modules that aren't xcframeworks. If we have any we exclude arm64 sim and spit out a warning about the modules and that it may fail. If we know we're running on an arm64 build of node, we error out with the message.

Reply:
Also note that I didn't have it smart enough to distinguish xcframeworks that are missing arm64 sim. That's a potential issue if people build a native module on Xcode 11 w/ the xcframework stuff landed. I'll need to see if we can't do more there.